python-avd/pyavd/_eos_designs/structured_config/metadata/cv_pathfinder.py

```diff
@@ -28,118 +29,78 @@ def _cv_pathfinder(self: AvdStructuredConfigMetadataProtocol) -> dict | None:
-                "interfaces": self._metadata_interfaces(),
-                "pathgroups": self._metadata_pathgroups(),
-                "regions": self._metadata_regions(),
-                "vrfs": self._metadata_vrfs(),
-            }
-
-        # Edge or transit
-        return {
-            "role": self.shared_utils.cv_pathfinder_role,
-            "ssl_profile": self.shared_utils.wan_stun_dtls_profile_name,
-            "vtep_ip": self.shared_utils.vtep_ip,
-            "region": region_name,
-            "zone": self.shared_utils.wan_zone["name"],
-            "site": site_name,
-            "interfaces": self._metadata_interfaces(),
-            "pathfinders": self._metadata_pathfinder_vtep_ips(),
-        }
-
-    def _metadata_interfaces(self: AvdStructuredConfigMetadataProtocol) -> list:
-        return [
-            {
-                "name": interface["name"],
-                "carrier": carrier["name"],
-                "circuit_id": interface.get("wan_circuit_id"),
-                "pathgroup": carrier["path_group"],
-                "public_ip": str(interface["public_ip"]) if self.shared_utils.is_cv_pathfinder_server else None,
-            }
-            for carrier in self.shared_utils.wan_local_carriers
-            for interface in carrier["interfaces"]
-        ]
-
-    def _metadata_pathgroups(self: AvdStructuredConfigMetadataProtocol) -> list:
-        return [
-            {
-                "name": pathgroup.name,
-                "carriers": [
-                    {
-                        "name": carrier.name,
-                    }
-                    for carrier in self.inputs.wan_carriers
-                    if carrier.path_group == pathgroup.name
-                ],
-                "imported_carriers": [
-                    {
-                        "name": carrier.name,
-                    }
-                    for carrier in self.inputs.wan_carriers
-                    if carrier.path_group in [imported_pathgroup.remote for imported_pathgroup in pathgroup.import_path_groups]
-                ],
-            }
-            for pathgroup in self.inputs.wan_path_groups
-        ]
-
-    def _metadata_regions(self: AvdStructuredConfigMetadataProtocol) -> list:
+            self.structured_config.metadata.cv_pathfinder._update(
+                role=self.shared_utils.cv_pathfinder_role,
+                ssl_profile=self.shared_utils.wan_stun_dtls_profile_name,
+                vtep_ip=self.shared_utils.vtep_ip,
+                region=region_name,
+                site=site_name,
+                address=self.shared_utils.wan_site.location if self.shared_utils.wan_site is not None else None,
+            )
+            self._metadata_interfaces()
+            self._metadata_pathgroups()
+            self._metadata_regions()
+            if self.structured_config.router_adaptive_virtual_topology.vrfs and self.structured_config.router_path_selection.load_balance_policies:
+                self._metadata_vrfs()
```

Comment:
Keep the self._metada_vrfs() only here and in the nested fiunction return early
```suggestion
            self._metadata_vrfs()
```

Reply:
Done